cocos/animation/marionette/pose-graph/x-node.ts

```diff
@@ -0,0 +1,43 @@
+import { VarInstance } from '../variable';
+import { PoseGraphNode } from './foundation/pose-graph-node';
+import { PoseGraphType } from './foundation/type-system';
+
+type Outputs = unknown[];
+
+export abstract class XNode extends PoseGraphNode {
```

Comment:
`ValueNode` does not distinguish these nodes from pose nodes: pose nodes also yield values.

Reply:
But PoseNode returns a Pose, doesn't it? The return values of Pose node and these value nodes don't feel like they have anything in common. And if Pose Node does have any similarity to XNode, maybe it would be better to have PoseNode inherit from ValueNode ?

Reply:
Now it's changed to be non-intrusive: https://github.com/cocos/cocos-engine/pull/14952/commits/31291aa8b9a7a62d1181621d4ae746a474b0b8a7

Reply:
The term "x node" is now renamed as "pure value node": https://github.com/cocos/cocos-engine/pull/14952/commits/f3b5c17bb22c0fd5b63e9a20b010b7cdb78262e0